### PR TITLE
Option to clear Sync server data and local browser state

### LINKED ISSUE #7275: Add a way to go back to add a sync code
I accidentally went into create a new sync account, and then had no way to go back to sync this device to another one. Please make a way to go back to that main screen and reset sync back to initial state if the user does the wrong selection like that.

cc @diracdeltas @diracdeltas

### PR DESCRIPTION
**Pull request by ayumi**
Fix #7275

Fix https://github.com/brave/sync/issues/51

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).

Test Plan:
1. Spawn a clean pyramid 0. Enable Sync and add bookmarks.
2. Spawn a clean pyramid 1 (`npm run start2`). Add it to 0's Sync profile and see the bookmarks sync over.
3. Save the Sync code words. We'll use this to demonstrate data was purged.
4. In Pyramid 0, go to Preferences > Sync and click Reset data. In the console you should see "Deleting user!!".
5. Close Pyramid 1, delete its user data, and spawn it clean again. Using the saved code words, try to re-add it to 0's deleted Sync profile. No bookmarks should sync over.
6. Back in Pyramid 0, you can use Sync freshly. Create a new profile and you should get a new set of code words.

Pix
screen shot 2017-02-16 at 16 48 00
screen shot 2017-02-16 at 16 48 15

### CODE REVIEW

File: app/renderer/components/preferences/syncTab.js [hunk 1]
```diff
@@ -242,6 +274,14 @@ class SyncTab extends ImmutableComponent {
           </div>
           : null
       }
+      {
+        this.isSetup
+          ? <div class='syncClearData'>
```
Comment (diracdeltas) [inline app/renderer/components/preferences/syncTab.js]: i think react is throwing an error because it expects className here

File: app/sync.js [hunk 2]
```diff
@@ -250,10 +254,20 @@ module.exports.onSyncReady = (isFirstRun, e) => {
...
   // sent by about:preferences when sync should be reloaded
   ipcMain.on(RELOAD_MESSAGE, () => {
     process.emit(RELOAD_MESSAGE)
   })
+  // sent by about:preferences when resetting sync
+  ipcMain.on(RESET_SYNC, (e) => {
+    if (dispatcherCallback) {
+      dispatcherCallback({actionType: syncConstants.SYNC_DELETE_USER})
+    }
+    appActions.changeSetting(settings.SYNC_ENABLED, false)
```
Comment (diracdeltas) [inline app/sync.js]: i would wait a short amount of time to call this to make sure that SYNC_DELETE_USER gets processed before the extension is disabled

Comment (ayumi) [inline app/sync.js]: could also add an IPC message for DELETED_USER and wait for that

**Comment by diracdeltas** [comment]
test plan worked for me. :+1:

File: app/sync.js (hunk 2 above)
Comment (diracdeltas) [inline app/sync.js]: that works too

**Comment by ayumi** [comment]
@diracdeltas have addressed comments.

deleting data requires sync to be enabled because it needs a valid AWS credential to issue the delete request. i've added a little message to explain:

screen shot 2017-02-21 at 21 29 03

**Pull request merged**